paragraph/scheduling/logger.h

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef PARAGRAPH_SCHEDULING_LOGGER_H_
+#define PARAGRAPH_SCHEDULING_LOGGER_H_
+
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "absl/status/status.h"
+#include "paragraph/scheduling/instruction_fsm.h"
+
+namespace paragraph {
+
+class Logger {
+ public:
+  // Logger factory that creates a Logger object given a filename with default
+  // empty string. If non-empty file name is given, Logger opens file to append
+  // to it. If it fails, it return fail status.
+  static shim::StatusOr<std::unique_ptr<Logger>> Create(
+      const std::string& filename = "");
+  ~Logger() = default;
+
+  // Flushes log to file and closes file
+  void FlushToFile();
```

Comment:
Why would a user need to call this?

Reply:
It's part of Domino effect. I wanted to implement logger setting after scheduler is created. So that required covering a case when you had logger, but then decided to change the filename. But that required to close old file before opening a new one, or test would fail because not all information would be synchronized with both files. In the end of the day, I use it in test, so I want it to be public. In general, I don't think there is any harm in providing a way to speed up flushing io buffer to file.